eval_converters/helm/adapter.py

```diff
@@ -124,385 +73,233 @@ def get_main_metric_name(run_path: str, schema_path: str) -> str:
 		except Exception as e:
-			self.logger.warning(f"Error getting quantization: {e}")
-			precision = BitPrecision.none
-			method = Method.None_
-
-		quantization = Quantization(
-			bit_precision=precision,
-			method=method,
-		)
-		inference_settings = InferenceSettings(
-			quantization=quantization,
-			generation_args=GenerationArgs(
-				temperature=adapter_spec.temperature,
-				stop_sequences=adapter_spec.stop_sequences,
-			)
-		)
+			print(f'Error during conversion to unified schema in directory "{dir_path}": {e}')
+			return None
+	
+	def _get_correct_response(self, references: List['Reference']) -> Optional[str]:
+		"""Extracts the text of the first reference that has tags."""
+		for ref in references:
+			if ref.tags:
+				return ref.output.text
+		return None
+
+	def _extract_detailed_evaluation_info_for_samples(
+		self, request_states: List[RequestState]
+	) -> List[DetailedEvaluationResultsPerSample]:
+		"""
+		Extracts detailed evaluation information for each sample from the request states.
+		"""
+		results: List[DetailedEvaluationResultsPerSample] = []
 		
-		# 2. PromptConfig
-		# 2.1. PromptClass
-		prompt_class = detect_prompt_class(adapter_spec.method)
-
-		evaluation_results: List[EvaluationResult] = []
-		for request_state in scenario_state.request_states:
-			# 3. Instance
-			# 3.1. SampleIdentifier
-			sample_identifier = SampleIdentifier(
-				dataset_name=scenario_dict['name'],
-				hf_repo="", # FIXME: use HF repo if available
-				hf_split=detect_hf_split(request_state.instance.split),
-				hf_index=-1,  # FIXME: use actual index if available
-			)
-			
-			# Extract ground truth: the first correct reference
-			# FIXME: need to modify the schema to support evaluation with more than one ground truth: https://crfm-helm.readthedocs.io/en/latest/code/#adding-new-scenarios
-			references = request_state.instance.references
-			ground_truth = {}
-			for i, ref in enumerate(references):
-				if "correct" in ref.tags:
-					ground_truth = {
-						"id": str(i),
-						"text": ref.output.text,
-					}
-					break
-			
-			# 3.2. ClassificationFields (required for classification tasks)
-			classification_fields = {}
-			output_mapping_dict = request_state.output_mapping or {}
-			if prompt_class == PromptClass.MultipleChoice: 
-				choices = [{"id": k, "text": v} for k, v in output_mapping_dict.items()]
-
-				classification_fields = {
-					"full_input": request_state.request.prompt,
-					"question": request_state.instance.input.text,
-					"choices": choices,
-					"ground_truth": ground_truth,
-				}
+		for state in request_states:
+			references = state.instance.references or []
+			correct_response = self._get_correct_response(references)
+
+			ground_truth = None
+			if correct_response:
+				ground_truth = next(
+					(
+						choice 
+						for choice, response in state.output_mapping.items() 
+						if response in correct_response
+					),
+					None
+				)
+
+			choices_list = [
+				f'{choice}. {response}' 
```

Comment:
This matches the string template for choices for our production leaderboard, but In theory, a HELM user could configure the rendering of the choices to be different from this.